app/controllers/users/reactivate_profile_controller.rb

```diff
@@ -19,7 +19,12 @@ def create
     def build_reactivate_profile_form
       ReactivateProfileForm.new(
         current_user,
-        params[:reactivate_profile_form].permit(:recovery_code, :password)
+        params[:reactivate_profile_form].
+        permit({:recovery_code => []}, :password).
+        tap do |obj|
+          code = obj[:recovery_code]
+          obj[:recovery_code] = code.join(' ') unless obj[:recovery_code].nil?
```

Comment:
We can probably just move the `code.join(' ')` inside the `ReactivateProfileForm` object?

Also I bet we could change the `permit` line to remove the extra curly braces:

```
permit(:password, recovery_code: [])
```